web/modules/custom/joinup_licence/src/Controller/LicenceComparerController.php

```diff
@@ -264,13 +290,69 @@ protected function buildRow(string $category, string $label, array $items): arra
       ];
     }
 
-    $this->padWithEmptyCells($row, [
+    $this->fillContentRows($row);
+
+    return $row;
+  }
+
+  /**
+   * Fills in the header rows with an add licence option and empty cells.
+   *
+   * @param array $row
+   *   A row array.
+   */
+  protected function fillHeaderRows(array &$row): void {
+    $amount = LicenceComparerHelper::MAX_LICENCE_COUNT - count($this->licences);
+    if ($amount === 0) {
+      return;
+    }
+
+    $classes = ['licence-comparer__header', 'licence-comparer__empty'];
+    // \Drupal\Core\Render\Element\Select::processSelect does not seem
+    // to be called if the element is not rendered through a form
+    // builder. Thus, the name of the field and the default values are
+    // not set. Manually set the additional properties.
+    $row[] = [
+      'data' => [
+        'licence_search' => [
+          '#type' => 'select',
+          '#title' => $this->t('Add licence'),
+          '#title_display' => 'invisible',
+          '#options' => ['' => $this->t('- Add licence -')]
+          + $this->getLicenceOptions(),
+          '#default_value' => '',
+          '#attributes' => [
+            'class' => ['auto_submit'],
+            'name' => 'Add licence',
```

Comment:
It's unusual to add a human-readable string as an element `name`. The `name` attribute is a machine-readable identifier. I would go with the `licence_search`, which it would have been the processed name if we were go with a form.